### PR TITLE
Upgraded dependencies to support Flutter 3.16.0

### PR DESCRIPTION
**Pull request by aryan-more**
Made it compatible with Flutter 3.16.0

### CODE REVIEW

**Comment by aryan-more** [comment]
Also [flutter-date-range-picker](https://github.com/linagora/flutter-date-range-picker/) dependency need to be upgraded .So i have also pushed  pull request for it

**Comment by dab246** [comment]
We should upgrade to the latest version `3.16`. If possible please help us with this @aryan-more

**Comment by aryan-more** [comment]
I just realized that I hadn't updated my Flutter SDK. So now, I have started figuring out which transitive dependencies are causing the build to fail.

**Comment by chibenwa** [comment]
Huge effort! Thanks a lot @aryan-more !

**Comment by aryan-more** [comment]
Hey @dab246 and @chibenwa, I have resolved the build issue on Flutter 3.16.0. The error/conflict was caused by a non-direct dependency, enough_platform_widgets, and as a result, I had to update some packages that required the enough_platform_widgets package.

I have forked the affected packages, patched them to support Flutter 3.16, and sent pull requests. Additionally, I have updated the Git URL in the pubspec.yaml file to point to these patched forks.

I have also sent pull requests to all of the following Git repositories. Please accept those pull requests.
[flutter-date-range-picker](https://github.com/linagora/flutter-date-range-picker/pull/9)
[enough_html_editor](https://github.com/linagora/enough_html_editor/pull/24)
[rich-text-composer](https://github.com/linagora/rich-text-composer/pull/23)

**Comment by aryan-more** [comment]
Hey @dab246 and @chibenwa I'm waiting for your reply

**Comment by hoangdat** [comment]
Hi @aryan-more , srr for late response, we are trying to release a stable version of `tmail` before more focus to upgrade with new Flutter. ANW, I commented on PRs.
Thanks and BRs.

**Comment by dab246** [comment]
> Hey @dab246 and @chibenwa, I have resolved the build issue on Flutter 3.16.0. The error/conflict was caused by a non-direct dependency, enough_platform_widgets, and as a result, I had to update some packages that required the enough_platform_widgets package.
> 
> I have forked the affected packages, patched them to support Flutter 3.16, and sent pull requests. Additionally, I have updated the Git URL in the pubspec.yaml file to point to these patched forks.
> 
> I have also sent pull requests to all of the following Git repositories. Please accept those pull requests. [flutter-date-range-picker](https://github.com/linagora/flutter-date-range-picker/pull/9) [enough_html_editor](https://github.com/linagora/enough_html_editor/pull/24) [rich-text-composer](https://github.com/linagora/rich-text-composer/pull/23)

We created an `upgrade_v3.16.0` branch on dependencies so we can test this version before merging into the master branch. You can create a PR to merge into it.

**Comment by dab246** [comment]
> Hey @dab246 and @chibenwa, I have resolved the build issue on Flutter 3.16.0. The error/conflict was caused by a non-direct dependency, enough_platform_widgets, and as a result, I had to update some packages that required the enough_platform_widgets package.
> 
> I have forked the affected packages, patched them to support Flutter 3.16, and sent pull requests. Additionally, I have updated the Git URL in the pubspec.yaml file to point to these patched forks.
> 
> I have also sent pull requests to all of the following Git repositories. Please accept those pull requests. [flutter-date-range-picker](https://github.com/linagora/flutter-date-range-picker/pull/9) [enough_html_editor](https://github.com/linagora/enough_html_editor/pull/24) [rich-text-composer](https://github.com/linagora/rich-text-composer/pull/23)

@aryan-more  We've merged your PRs into the `upgrade_v3.16.0` branch. Please use this branch for dependencies that need to be changed on TMail's flutter `3.16.0`.

File: pubspec.yaml
Comment (hoangdat) [inline pubspec.yaml:92]: can you make the PR to https://github.com/linagora/dns_client. Thanks and BRs

Comment (aryan-more) [reply pubspec.yaml:92]: Done

Comment (hoangdat) [reply pubspec.yaml:92]: merged, please change this

Comment (aryan-more) [reply pubspec.yaml]: @hoangdat merged

**Review by hoangdat — APPROVED** [review]

**Comment by hoangdat** [comment]
hi @aryan-more , I updated branch [linagora:upgrade/upgrade_v3.16.0](https://github.com/linagora/tmail-flutter/tree/upgrade/upgrade_v3.16.0), please try to resolve the conflict.

Thanks and BRs

**Comment by hoangdat** [comment]
Hi @aryan-more please use `rebase` instead of merge. Anw, still have the conflict :(

**Comment by aryan-more** [comment]
@hoangdat, I'm able to run it locally. However, the GitHub Pages has this error: 'Unable to fetch the information for the environment 'PR-2341' targeted by this job.' I think there is an error regarding the environment variable.

**Comment by hoangdat** [comment]
> @hoangdat, I'm able to run it locally. However, the GitHub Pages has this error: 'Unable to fetch the information for the environment 'PR-2341' targeted by this job.' I think there is an error regarding the environment variable.

ok for the run, ignore it. But please use `rebase` instead of `merge`. We want to keep the linear history. Thanks

**Comment by aryan-more** [comment]
@hoangdat i was unaware of this difference between merge and rebase.Should i roll back to previous commit and then rebase

**Comment by dab246** [comment]
> @hoangdat i was unaware of this difference between merge and rebase.Should i roll back to previous commit and then rebase

Thank you for your contribution. We will help you rebase it.

**Comment by dab246** [comment]
We have updated flutter version `3.16.0` upgrade in PR (https://github.com/linagora/tmail-flutter/pull/2380). All issues will be discussed at that PR. Please close this PR. Thanks.

**Pull request closed without merging**
